extensions/gson/src/main/java/io/jsonwebtoken/io/GsonDeserializer.java

```diff
@@ -0,0 +1,42 @@
+package io.jsonwebtoken.io;
```

Comment:
Because of the Java >= 9 module system, separate implementation .jars cannot export the same packages as the base api .jar.  As a result we have to fix this for the Jackson and org.json implementations and that's being tracked under #399 .

But because this gson extension is new code, we can avoid this mistake and do it properly from the start.  As a result, please move these to a `io.jsonwebtoken.gson.io` package.

This is to use a convention moving forward:

`io.jsonwebtoken` prefix + `.<extension-name>.` + `<mirrored suffix in the api module>`

For example, if there is an interface in the API module in the `io.jsonwebtoken.foo` package, and an extension named `bar` wanted to implement it, then:

* the bar implementation would be in package `io.jsonwebtoken.bar.foo`, and
* the .java file would be in `extensions/bar/src/main/java/io/jsonwebtoken/bar/foo`

This works for the Java >= 9 module system because `io.jsonwebtoken.*` is exported from the api .jar, and only `io.jsonwebtoken.gson.*` is exported from the gson .jar.  There is no set intersection of packages between the two (which the JDK Module System requires).